File: python/secret-auction.py
auction = {}
new_bid = True
while new_bid:
  name = input("What is your name?: ")
  bid_amount = int(input("What's your bid?: $"))
  auction[name] = bid_amount
  restart = input("Are there any other bidders? Type 'y' or 'n' ")

  if restart == 'n':
    new_bid = False

max_amount = 0
bidder_name = ""
for name in auction:
  if auction[name] > max_amount:
    max_amount = auction[name]
    bidder_name = name

print(f"The winner is {bidder_name} with a bid of ${max_amount}")

File: go/secret-auction.go
package main

import (
	"fmt"
)

func main() {
	auction := make(map[string]int)
	new_bid := true
	for new_bid {
		fmt.Print("What is your name?: ")
		var name string
		fmt.Scanln(&name)
		fmt.Print("What's your bid?: $")
		var bid_amount int
		fmt.Scanln(&bid_amount)
		auction[name] = bid_amount

		fmt.Print("Are there any other bidders? Type 'y' or 'n' ")
		var restart string
		fmt.Scanln(&restart)

		if restart == "n" {
			new_bid = false
		}
	}

	var (
		max_amount  int
		bidder_name string
	)

	for key, value := range auction {
		if value > max_amount {
			max_amount = value
			bidder_name = key
		}
	}

	fmt.Printf("The winner is %s with a bid of $%d\n", bidder_name, max_amount)
}
